### PR TITLE
feat(cache): add name for cache instance

### PR DESCRIPTION
**Pull request by W-Mai**
### Description of the feature or fix

Add name for cache instance

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/misc/cache/lv_cache.h [hunk 1]
```diff
@@ -200,6 +200,21 @@ void   lv_cache_set_create_cb(lv_cache_t * cache, lv_cache_create_cb_t alloc_cb,
  * @param user_data     A user data pointer.
  */
 void   lv_cache_set_free_cb(lv_cache_t * cache, lv_cache_free_cb_t free_cb, void * user_data);
+
+/**
+ * Give a name for a cache object.
```
Comment (kisvegabor) [inline src/misc/cache/lv_cache.h]:
```suggestion
 * Give a name for a cache object. Only the pointer of the string is saved.
```

**Review by kisvegabor — COMMENTED** [review]
Looks good, but I wonder why did you need this feature.

**Comment by W-Mai** [comment]
> Looks good, but I wonder why did you need this feature.

I want to add more debug features to lvgl:

1. Module independent log capability switch

2. Module independent profile capability switch

3. cache monitor, etc.

Among them, the cache monitor needs to name the cache, so that it can more clearly monitor the work of each cache in lvgl. 😁

**Review by kisvegabor — APPROVED** [review]
Okay, thanks! Approved :slightly_smiling_face:

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**